casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -141,6 +150,44 @@ object Validate {
       }
     }
 
+  /**
+    * Validate no deploy by the same (user, millisecond timestamp)
+    * has been produced in the chain
+    */
+  def repeatDeploy[F[_]: Monad: Log: BlockStore](block: BlockMessage,
+                                                 genesis: BlockMessage,
+                                                 dag: BlockDag): F[Boolean] =
+    BlockStore[F].asMap().flatMap { internalMap: Map[BlockHash, BlockMessage] =>
+      {
+        val deployKeySet = block.getBody.newCode.map(d => (d.getRaw.user, d.getRaw.timestamp)).toSet
+        val iterator = DagOperations.bfTraverse(Some(genesis))(x =>
```

Comment:
Can we use b (for "block") instead of x?

Reply:
done